x-pack/plugins/inference/common/ensure_multi_turn.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Message, MessageRole } from './chat_complete';
+
+function isUserMessage(message: Message): boolean {
+  return message.role !== MessageRole.Assistant;
```

Comment:
Is `MessageRole.Tool` considered a user message then?

Reply:
Yes, it is a reply from the user's system.